plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPlugin.java

```diff
@@ -193,11 +193,12 @@ public String NabuToken() throws RuntimeException {
   private Optional<String> findMetadataTextValue(
       JsonNode metadata, String bestKey, String backupKey) {
     // metadata for earlier case archives were submitted with snake case keys
-    if (metadata.findValue(bestKey) != null) {
-      return Optional.of(metadata.findValue(bestKey).textValue());
+    if (metadata.findValue(bestKey) != null && metadata.findValue(bestKey).textValue() != null) {
```

Comment:
```suggestion
    if (metadata.findValue(bestKey) != null) {
```
If you're using Optional.ofNullable, it will create an Optional.empty if metadata.findValue(bestKey).textValue() is null. 
Ditto for line 199.